### DISCUSSION TITLE
LOIJ 1020求質數 Runtime Error

### DISCUSSION THREAD

**Discussion by jeffw858** [Lidemy/forum#273] [程式導師實驗計畫]
1. 起初我在上傳的時候一直出現runtime error請問這邊的獲取資料應該如何宣告呢？
2. 我後來有上網找到宣告老師在課堂中沒有說的arr部分，在vs可以跑出正解但是上傳時卻寫WA，請問是哪裡出問題呢？(附上程式碼)
<img width="733" alt="截圖 2022-07-26 下午8 10 50" src="https://user-images.githubusercontent.com/110077472/181152526-97fdbdea-edcd-43f7-a295-2368f689c409.png">

**Comment by cwc329** [comment]
```js
var arr = Array.from({length: number}, (_, index) => index + 1);
```
這段有點怪怪的，
這樣你的 array 會是固定的模式，
但是題目的 array 應該是要從 input 去讀。

**Reply by jeffw858** [reply]
C大您好，但是我在length有用n作為長度這樣不會讀到input嗎

**Comment by nuoxoxo** [comment]
Hi
```j
var arr = Array.from({length: number}, (_, index) => index + 1);
```
這一行的意思是：生成一個長度為 `Number(lines[0])` 、差為 1 的等差陣列
`solve( )` 並沒有處理測試資料，而是處理了這個等差陣列
其實你的 input 通過 1 - 13 行已經放進 lines ，你要處理的是 lines

**Reply by jeffw858** [reply]
Hi n大，可以讓我伸手牌一下告訴我這裡應該要怎麼寫才是正確的嗎？我不知道要怎麼把lines的資料放入陣列

**Reply by nuoxoxo** [reply]
> 要怎麼把lines的資料放入陣列

你已經把lines的資料放入陣列。再提醒就會是答案，題就不是你做的了 : D

**Reply by jeffw858** [reply]
我就是已經想不到了QQ
試過arr= Number(lines[1])、arr=[n]、arr=[ ]等。想說到底要怎麼先宣告arr出來

**Reply by aszx87410** [reply]
你其實從頭到尾都不需要 arr 這東西，因為輸入已經在 lines 裡面了，而且 lines 已經是陣列，你直接對 lines 做 for loop 就好

**Reply by jeffw858** [reply]
![截圖 2022-07-27 下午7 25 54](https://user-images.githubusercontent.com/110077472/181406870-691dc425-b96e-4a99-8ab3-fed523501413.png)
a大您好，我其實也試過我其實也試過在函數裡面直接用n去跑，但是依然runtime error...我能想到的方法都試過了...

**Reply by aszx87410** [reply]
``` js
let n = Number(lines[0])
for(let i=0; i<n; i++) {
  if (isPrime(Number(lines[i+1]))) {
  // ...
}
```

**Reply by jeffw858** [reply]
原來還有這招！！！！感謝您！！！！！我一直以為只能用陣列去跑
